### PR TITLE
Semver class

### PR DESCRIPTION
**Pull request by chillaq**
# Ruby SDK

### CODE REVIEW

File: spec/engine/matchers/semver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
+require 'spec_helper'
+
+describe SplitIoClient::Semver do
+  let(:valid_versions) { ["1.1.2", "1.1.1", "1.0.0", "1.0.0-rc.1", "1.0.0-beta.11", "1.0.0-beta.2",
```
Comment (sanzmauro) [inline spec/engine/matchers/semver_spec.rb:4]: Why are you not using the csv example? 
The main idea of the csv file is to have the same tests across the SDKs, and if we want to update the file or add more examples, the changes are easier.

File: lib/splitclient-rb/engine/matchers/semver.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    #
+    # @param version [String] raw version as read from splitChanges response.
+    def initialize(version)
+      @major = 0
+      @minor = 0
+      @patch = 0
+      @pre_release = []
+      @is_stable = false
+      @old_version = version
+      parse
```
Comment (sanzmauro) [inline lib/splitclient-rb/engine/matchers/semver.rb:22]: please don't parse at constructor level.
if you have a `RuntimeError` at this level you are not catching it.

**Review by sanzmauro — CHANGES_REQUESTED** [review]

File: spec/engine/matchers/semver_spec.rb (hunk 1 above)
Comment (chillaq) [reply spec/engine/matchers/semver_spec.rb]: will do

File: lib/splitclient-rb/engine/matchers/semver.rb (hunk 2 above)
Comment (chillaq) [reply lib/splitclient-rb/engine/matchers/semver.rb:18]: will do

**Review by sanzmauro — APPROVED** [review]

**Pull request merged**